peer.go

```diff
@@ -263,7 +265,7 @@ var _ lnpeer.Peer = (*peer)(nil)
 // pointer to the main server. It takes an error buffer which may contain errors
 // from a previous connection with the peer if we have been connected to them
 // before.
-func newPeer(conn net.Conn, connReq *connmgr.ConnReq, server *server,
+func newPeer(cfg *Config, conn net.Conn, connReq *connmgr.ConnReq, server *server,
```

Comment:
this is fine for now, @Crypt-iQ is also working on refactoring this area, we can take that opportunity to pass in a more minimal subset of the config